### PR TITLE
Webauthn support in many

### PR DESCRIPTION
**Pull request by hansl**
This adds an Identity for WebAuthn in `many-identity-webauthn`. This works with the server perfectly.

### CODE REVIEW

**Comment by hansl** [comment]
@fmorency I cannot seem to get bazel to like this PR for some reasons. I get the following error:

```
error[E0432]: unresolved import `many_identity_webauthn::WebAuthnIdentity`
  --> src/many/src/main.rs:11:5
   |
11 | use many_identity_webauthn::WebAuthnIdentity;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `WebAuthnIdentity` in the root
```

Seems like the feature isn't enabled? Ideas?

**Comment by fmorency** [comment]
> @fmorency I cannot seem to get bazel to like this PR for some reasons. I get the following error:
> 
> ```
> error[E0432]: unresolved import `many_identity_webauthn::WebAuthnIdentity`
>   --> src/many/src/main.rs:11:5
>    |
> 11 | use many_identity_webauthn::WebAuthnIdentity;
>    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `WebAuthnIdentity` in the root
> ```
> 
> Seems like the feature isn't enabled? Ideas?

There's support for that feature in my Bazel Docker branch. It should allow you to

`--//src/many-ledger:webauthn_testing=1`

See https://github.com/fmorency/many-framework/blob/2f28a04f709ddbc8517a5d2b682198eaaa1a8f46/src/many-ledger/BUILD.bazel#L18-L21

**Comment by hansl** [comment]
@fmorency Just added the feature in bazel. No downsize to over compile in Bazel I think.

File: src/many-identity-webauthn/src/identity.rs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                .map(2)?
+                .u8(0)?
+                .bytes(&protected_header.to_vec().unwrap())?
+                .u8(1)?
+                .str(&base64::encode(hash))?;
+            Ok(encoder.into_writer())
+        })();
+        let challenge = challenge.unwrap();
+
+        // let mut provider = u2fhid::U2FHid::new();
```
Comment (fmorency) [inline src/many-identity-webauthn/src/identity.rs:87]: Remove

File: src/many-identity-webauthn/src/identity.rs [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        }
+
+        let hash = hash.finalize().to_vec();
+
+        let protected_header = envelope.protected.clone();
+        let challenge: Result<Vec<u8>, minicbor::encode::Error<_>> = (|| {
+            let mut encoder = minicbor::Encoder::new(Vec::new());
+            encoder
+                .map(2)?
+                .u8(0)?
```
Comment (fmorency) [inline src/many-identity-webauthn/src/identity.rs:79]: nit: Use a constant. Same with the other one below.

File: src/many-identity-webauthn/src/identity.rs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+            Ok(encoder.into_writer())
+        })();
+        let challenge = challenge.unwrap();
+
+        // let mut provider = u2fhid::U2FHid::new();
+        let mut provider = u2fhid::U2FHid::new();
+
+        let public_key = PublicKeyCredentialRequestOptions {
+            challenge: challenge.into(),
+            timeout: Some(60_000),
```
Comment (fmorency) [inline src/many-identity-webauthn/src/identity.rs:92]: nit: 
ONE_SECOND = 1000
ONE_MINUTE = 60 * ONE_SECOND

File: src/many-identity-webauthn/src/identity.rs [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+                ]),
+            }],
+            extensions: None,
+        };
+
+        let r = provider
+            .perform_auth(
+                Url::parse(self.origin_url.as_str()).unwrap(),
+                public_key,
+                60_000,
```
Comment (fmorency) [inline src/many-identity-webauthn/src/identity.rs:111]: Idem. Make a const with that value.

File: src/many-identity-webauthn/src/identity/u2fhid.rs [hunk 5]
```diff
@@ -0,0 +1,421 @@
+//! This is a copy of the webauthn-authenticator-rs u2fhid implementation,
+//! with a difference in the API; the perform_auth function returns a
+//! PublicKeyCredentials which has no rawId in it, as the Ledger Nano
+//! doesn't have proper CTAP 2.1 support (notably the attestation object
+//! is missing).
+//! We only use Attestation to get the rawId, and we (MANY) don't use the
+//! rawId at all, so skipping it does nothing and fixes a support bug.
```
Comment (fmorency) [inline src/many-identity-webauthn/src/identity/u2fhid.rs:7]: Interesting, can you link to some issue/documentation about this?

Also, can you highlight the changes you made in this file?

File: src/many-identity-webauthn/src/verifier.rs [hunk 6]
```diff
@@ -0,0 +1,473 @@
+use coset::cbor::value::Value;
```
Comment (fmorency) [inline src/many-identity-webauthn/src/verifier.rs:1]: Did you change anything at all in this file? It looks like a copy of my code, and I won't review it if no change has been made.

**Review by fmorency — CHANGES_REQUESTED** [review]
Thanks!
Some comments, but nothing major.

The `rp` for the Manifest Network would be the Alberto URL?

File: src/many-identity-webauthn/src/identity.rs (hunk 2 above)
Comment (hansl) [reply src/many-identity-webauthn/src/identity.rs:79]: Moved to a special type. Simpler.

File: src/many-identity-webauthn/src/verifier.rs (hunk 6 above)
Comment (hansl) [reply src/many-identity-webauthn/src/verifier.rs:1]: No, just moved from `lib.rs`.

Comment (hansl) [reply src/many-identity-webauthn/src/verifier.rs:1]: Actually now I just did with the Challenge type.

**Review by fmorency — APPROVED** [review]
Thanks! LGTM assuming CI is fixed.

**Pull request merged**